common/src/main/java/com/wynntils/models/character/type/SetInfo.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.character.type;
+
+import net.minecraft.world.item.ItemStack;
+
+public class SetInfo {
```

Comment:
At first I thought "SetInfo" was what you called "SetInstance", but it seems you are using the names the other way around. For me, "SetInfo" sounds like a name for a generic information about the sets that exists in Wynncraft, while a SetInstance might be an actual instance that you are currently wearing.

But the purpose of this class seem to be only to handle the duplicate ring issue, right? Then I suggest the name should be more direct, like DuplicateSetItemDetector.

Reply:
I left it as SetInstance. The class also collects some item relevant to a set so it can be parsed later.